src/osd/PG.cc

```diff
@@ -4184,13 +4192,15 @@ void PG::chunky_scrub(ThreadPool::TPHandle &handle)
              p != pg_log.get_log().log.rend();
              ++p) {
           if (cmp(p->soid, scrubber.start, get_sort_bitwise()) >= 0 &&
-	      cmp(p->soid, scrubber.end, get_sort_bitwise()) < 0) {
+	      cmp(p->soid, scrubber.end, get_sort_bitwise()) <= 0) {
```

Comment:
The original PR didn't have the change because the later release would never have to interact with Hammer OSDs.  Hammer OSDs had a small bug for EC pools http://tracker.ceph.com/issues/17491 that would include the end element.  See comment in PG.h.